lib/origen_testers/smartest_based_tester/base/flow.rb

```diff
@@ -264,6 +264,25 @@ def capture_lines
           @lines_buffer.pop
         end
 
+        def on_loop(node)
+          loop_start = node.children[0]
+          loop_end = node.children[1]
+          loop_inc = node.children[2]
+          index_flag = node.children[3] if node.children[3].is_a?(String)
+          if index_flag.is_a?(String)
+            line "for @#{index_flag} = #{loop_start}; @#{index_flag} < #{loop_end} ; @#{index_flag} = @#{index_flag} + #{loop_inc}; do"
+          else
+            line "for @index = #{loop_start}; @index < #{loop_end} ; @index = @index + #{loop_inc}; do"
```

Comment:
I think you need to generate something more unique that just `@index`, e.g. consider the case of nested loops.
If you look around in here there should be other examples of generating unique ids.

Reply:
@ginty Is that the responsibility of Origen?  Seems like that should be up to the user similar to how one would use different variables in a C for loop.

Reply:
Yes, this is the kind of detail that a framework should take care of

Reply:
Also consider the Origen API is trying to be as ATE-agnostic as possible. If a user only wants to loop X times, then they should be able to express that without worrying about flow vars. Such things may not even make sense on other platforms.

Reply:
> Also consider the Origen API is trying to be as ATE-agnostic as possible. If a user only wants to loop X times, then they should be able to express that without worrying about flow vars. Such things may not even make sense on other platforms.

The current ATP loop API allows the user to pass an integer or a string representing a flow variable for the `to` option.  Wouldn't we just handle that in each of the tester generators?  If a particular generator does not use an option available in the ATP flow method then it errors and instructs the user.

Reply:
This is the way Origen works man, with the current loop API I would and should expect this to just work:

~~~ruby
loop from: 1, to: dut.number_of_x, step: 1 do
    loop from: 1, to: dut.number_of_y, step: 1 do
      func :some_test
    end
end
~~~

If the v93k needs a flow var to be defined to enable that to work, then its Origen's responsibility to create one (if the user doesn't care about what it's called), and make sure what it is creating will work in cases like the above.